workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-etc-spring-boot/src/main/java/com/liferay/marketplace/ConsoleRestController.java

```diff
@@ -121,12 +121,68 @@ public void postProvisioning(
 	}
 
 	@PostMapping("uninstall-app/{orderId}")
-	public void uninstallApp(@PathVariable("orderId") long orderId)
+	public void uninstallApp(
+			@PathVariable("orderId") long orderId, @RequestBody String json)
 		throws Exception {
 
 		try {
 			_consoleService.uninstallApp(orderId);
 
+			Order order = _marketplaceService.getOrder(orderId);
+
+			Map<String, String> customFields =
+				(Map<String, String>)order.getCustomFields();
+
+			JSONObject jsonObject = new JSONObject(json);
+
+			JSONArray cloudProvisioningJSONArray = new JSONArray(
+				customFields.get("cloud-provisioning"));
+
+			for (int i = 0; i < cloudProvisioningJSONArray.length(); i++) {
+				JSONObject cloudProvisioningJSONObject =
+					cloudProvisioningJSONArray.getJSONObject(i);
+
+				if (!Objects.equals(
+						cloudProvisioningJSONObject.getLong("orderItemId"),
+						jsonObject.getLong("orderItemId"))) {
+
+					continue;
+				}
+
+				JSONArray deploymentsJSONArray =
+						cloudProvisioningJSONObject.getJSONArray("deployments");
+
+				for (int j = 0; j < deploymentsJSONArray.length(); j++) {
+					JSONObject deploymentJSONObject =
+						deploymentsJSONArray.getJSONObject(j);
+
+					if (Objects.equals(
+							deploymentJSONObject.get("projectId"),
+							jsonObject.get("projectId"))) {
```

Comment:
@welitonpe I think we should use `id` instead of `projectId` since it is unique

```suggestion
							deploymentJSONObject.get("id"),
							jsonObject.get("id"))) {
```

Reply:
To be clear I'm talking about this

```
[{"deployments":[{"createdAt":1727980576511,"orderId":35718368,"appId":"mxmdq","id":"grivtzphbibpdskckb","projectId":"extlxcv1e6-ext35717400"}],"quantity":1,"orderItemId":35718371,"shippedQuantity":1,"sku":"35620554v10ts"}]
```

"id": that contains "grivtzphbibpdskckb"
to be used instead of "projectId" that contains "extlxcv1e6-ext35717400"

Because the same CX can be installed multiple times inside the same project